### PR TITLE
Refactor player action, but state machine still not work

### PR DESCRIPTION
**Pull request by nonorze-VOXX**
state machine still not refactor

### CODE REVIEW

File: Assets/Script/Player/PastPlayer.cs [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
+            }
+
+            playerData.pastPlayerData.path.Enqueue(player.transform.position);
+        }
 
+        private void Init()
+        {
+            playerData.pastPlayerData.path = new Queue<Vector2>();
+            playerData.pastPlayerData.pathTime = 0;
+        }
```
Comment (kryr34) [inline Assets/Script/Player/PastPlayer.cs:36]: why don't just put in `Start()`

File: Assets/Script/boss/bossManage.cs [hunk 2]
```diff
@@ -100,25 +106,27 @@ private void Move()
...
         }
     }
+
     private void ChargeMove()
     {
-
         if (bossT.position.x > playerT.position.x)
             bossT.Translate(new Vector2(-chargeSpeed, 0));
         else
             bossT.Translate(new Vector2(-chargeSpeed, 0));
```
Comment (kryr34) [inline Assets/Script/boss/bossManage.cs:124]: why?

File: Assets/Script/boss/bossManage.cs [hunk 3]
```diff
@@ -100,25 +106,27 @@ private void Move()
...
             }
-            if(AttackLoopNum == 4)
+
+            if (AttackLoopNum == 4)
                 AttackLoop();
             else if (playerT.position.x - bossT.position.x > distance)
                 bossT.Translate(new Vector2(-speed, 0));
             else
                 AttackLoop();
         }
```
Comment (kryr34) [inline Assets/Script/boss/bossManage.cs:116]: can change to tuple switch case or [pattern](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#positional-pattern) FYI
or better if else statement
```cs
if ( isA ) {}
else if (isB and isC) {}
else if (isC) {}
else {}
```

File: Assets/Script/Player/PlayerManager.cs [hunk 4]
```diff
@@ -1,180 +1,216 @@
...
+                    break;
+                }
+                case false when playerData.jumpData.CdCounter > playerData.jumpData.Cd:
+                    playerData.stateCheck.Jump = true;
+                    playerData.jumpData.CdCounter = 0;
+                    break;
+                case false:
+                    playerData.jumpData.CdCounter += Time.deltaTime;
+                    break;
+            }
```
Comment (kryr34) [inline Assets/Script/Player/PlayerManager.cs]: also this, if-else statement maybe better

File: Assets/Script/Player/PlayerManager.cs [hunk 5]
```diff
@@ -1,180 +1,216 @@
...
-            && collision.gameObject.transform.rotation == this.transform.rotation)
+        private bool GetRightKey()
         {
-            data.enemyHp -= data.playerDamege;
-            attacked = true;
-            if (data.enemyHp <= 0)
-            {
-            }
+            return Input.GetKey(KeyCode.D) || Input.GetKey(KeyCode.RightArrow);
         }
```
Comment (kryr34) [inline Assets/Script/Player/PlayerManager.cs]: I prefer use variable instead to function if no other references

**Review by kryr34 — COMMENTED** [review]
I didn't mention all parts which with same problem, please check whole file.

File: Assets/Script/boss/bossManage.cs (hunk 2 above)
Comment (nonorze-VOXX) [reply Assets/Script/boss/bossManage.cs:124]: @vhan9999

File: Assets/Script/boss/bossManage.cs (hunk 3 above)
Comment (nonorze-VOXX) [reply Assets/Script/boss/bossManage.cs:116]: @vhan9999

File: Assets/Script/Player/PlayerManager.cs (hunk 5 above)
Comment (nonorze-VOXX) [reply Assets/Script/Player/PlayerManager.cs]: ok

**Review by kryr34 — COMMENTED** [review]
Merciless refactoring or merge quickly

**Review by kryr34 — COMMENTED** [review]
Merciless refactoring or merge quickly

**Review by kryr34 — APPROVED** [review]
Merciless refactoring or merge quickly

**Pull request merged**